### ISSUE TITLE
investigate test_agreement function

### ISSUE THREAD

**Issue by oSamTo** [NERC-CEH/ukem#8]
The `test_agreement()` function has returned output:input ratios of ~0.97 for all GHGs for all years, which immediately raises suspicion.

Check the function over and the data steps in ukem.R, starting with (but not restricted to): 

- area of LL cells in m2 when applied to the emission m-2 compared to what is converted to LL in the first place 
- sector allocation

**Comment by oSamTo** [comment]
this is being done on the 'agreement_check' branch

**Comment by oSamTo** [comment]
`test_agreement()` has been altered to use `terra::cellSize` as `terra::area` is just `raster::area`. The new `cellSize()` function is a bit more accurate. 

However, leaving v_snap_id = 1:n_sectors in `test_agreement()` will now carry through 12 sectors (i.e. including marine) to be compared to the original `dt_ts_uk` which does not have marine. This is because we are presetting `v_sector` to include "marine".

Possible solution 1: include marine data into `getAlpha_byYear()`, thereby having a more complete record of what is going into the stacks
Possible solution 2: only run `test_agreement()` with `v_snap_id = 1:11` (which seems a bit of a shame)
Possible solution 3: add marine layer total to `F_total_ts` in `test_agreement()`

probably more options also

**Comment by oSamTo** [comment]
righto, think i've found the source of this! There isn't really a reason why the agreement factor (is that the right term?) should be <0.995 or > 1.005 or thereabouts for given gas in a given year, when excluding marine (so SNAPs 1:11). This is because the data is gathered, points are added, it's re-projected but crucially it then undergoes `scale_stack_to_nationalTotal()`. 

The slight discrepancy we are seeing is that `unit_conversion()` and `test_agreement()` are not treating the data the same: in the former it is divided by 1e6 to represent km-2 to m-2 while in the latter, it is multiplied by the actual area of the LL cell in m2. However a LL raster stack is passed to `unit_conversion()` so i guess the conversion should not be km-2 to m-2 but from cell-1 to m-2, much like `test_agreement()` but in reverse.

**Comment by peterlevy** [comment]
`terra::global(r, fun="sum", weights=w) ` would do the same thing, possibly faster.  I can't make out from the documentation whether this intrinsically does the area weighting, and the weights are only for additional weighting if needed for something else.

**Comment by oSamTo** [comment]
good point re `1e6 m^2 in 1 km^2` , that is not the bit to be adjusted. My comment refers to;

https://github.com/NERC-CEH/ukem/blob/15499f866e7bf48329536b548a21ec9d0ecca0f6/R/ukem.R#L629

in which the unit is always converted on the premise the emission is in BNG, through the argument `unit_in = "Mg / km^2 / yr"`

What you suggests works, I am pretty sure of it (without actually implementing it). I think either that 
a) `unit_in =` argument or the function `unit_conversion()` needs some adjusting/expanding as we are claiming units in are km^2 (when they're not) or  
b) just implement what you say, (i can now see your terra comment too)
